tests/test_range.py

```diff
@@ -789,6 +789,89 @@ def test_week_start(self):
                           datetime(2017, 6, 20),
                           week_start=0)
 
+    def test_weekend(self):
+        now = datetime.now()
+
+        # Frozen date is on a weekend
+        self.assert_range('this weekend',
+                          datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                          datetime(2017, 6, 19, WEEKEND_END_HOUR))
+
+        self.assert_range('this weekend',
+                          datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                          now,
+                          context=CONTEXT_PAST)
+
+        self.assert_range('this weekend',
+                          now,
+                          datetime(2017, 6, 19),
+                          context=CONTEXT_FUTURE)
+
+        self.assert_range('last weekend',
+                          datetime(2017, 6, 9, WEEKEND_START_HOUR),
+                          datetime(2017, 6, 12, WEEKEND_END_HOUR))
+
+        self.assert_range('next weekend',
+                          datetime(2017, 6, 23, WEEKEND_START_HOUR),
+                          datetime(2017, 6, 26, WEEKEND_END_HOUR))
+
+        self.assert_range('since this weekend',
+                          datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                          now)
+
+        self.assert_range('since last weekend',
+                          datetime(2017, 6, 9, WEEKEND_START_HOUR),
+                          now)
+
+        self.assert_range('until next weekend',
+                          now,
+                          datetime(2017, 6, 23, WEEKEND_START_HOUR))
+
+        # On a non-weekend
+        with freeze_time('2017-06-14 10:43:09'):
+            self.assert_range('this weekend',
+                              datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                              datetime(2017, 6, 19, WEEKEND_END_HOUR))
+
+            self.assert_range('this weekend',
+                              datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                              datetime(2017, 6, 19, WEEKEND_END_HOUR),
+                              context=CONTEXT_PAST)
+
+            self.assert_range('this weekend',
+                              datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                              datetime(2017, 6, 19, WEEKEND_END_HOUR),
+                              context=CONTEXT_FUTURE)
+
+            self.assert_range('last weekend',
+                              datetime(2017, 6, 9, WEEKEND_START_HOUR),
+                              datetime(2017, 6, 12, WEEKEND_END_HOUR))
+
+            self.assert_range('next weekend',
+                              datetime(2017, 6, 16, WEEKEND_START_HOUR),
+                              datetime(2017, 6, 19, WEEKEND_END_HOUR))
```

Comment:
next weekend is still the weekend of the next week if you are on a weekday datetime(2017, 6, 23, WEEKEND_START_HOUR), datetime(2017, 6, 26, WEEKEND_END_HOUR)

Reply:
That would be inconsistent with other cases of "next" and "this", like "this Saturday" == "next Saturday" if today is Friday.

Reply:
it is not inconsistent with your Saturday example because you have to look at the weekend as a 60 hour chunk of time.

On monday at 8 am, next weekend is not this weekend.

Reply:
You're missing my point. I can't comment inline but here is what I'm talking about: https://github.com/iAmPlus/timestring/pull/33/files#diff-f20349de99bcf4d3cf50ed65cd310325L518 "This Saturday" and "next Saturday" are currently the same day. I think this is how people use "this" and "next". Weekends should work the same.